### ISSUE TITLE
Integer signedness consistency

### ISSUE THREAD

**Issue by kvark** [gpuweb/gpuweb#2570]
This is a follow-up to Google's suggested ideas of VF2F in Feb 2022. It's an alternative proposal to the "Texture Built-in Function Integer Parameters" slide. We discussed it with @jimblandy and wanted to provide feedback before F2F to make our tomorrow's discussions more productive.

Current situation: API has different expectations for integer signs:
  - `workgroup_size` attribute is *unsigned*
  - `textureLoad` (and the level of `textureSample`) arguments are *signed*
  - builtin types, like `vertex_index`, `instance_index`, etc, are all *unsigned*
  - array indexing accepts both signs

One problem is that this model is internally inconsistent. I.e. if `textureLoad` expects coordinates from 0 to W-1, why is it signed? See #1257
Another problem that flow naturally from here is friction for the user. If API logic is inconsistent, users don't incorporate it into their mental models, and hence their code often clashes with API expectations.

If all of those areas of APIs (listed here) work with unsigned, why are we expecting the signed values in some of them? I believe the answer is split in 2 parts:
  1. Typing `1` is easier than `1u`. This is a really old argument, and it doesn't hold any ground with #2200. I know Google wants to revise this proposal, and it's not finished yet, etc. But I think we agree that we want it in V1 in some form, so this problem of unsigned literals should be considered done.
  2. Some computations naturally produce signed results, which we expect people to put into `textureSample` arguments. Our statement here is that `u32` isn't at disadvantage here. If the computations are done in `f32`, then just casting it to `u32` instead of `i32` in the end would take exactly the same effect. The bounds checking code works the same for -1 and 0xFFFFFFFF.

With this in mind, we think the spec should come to internal consistency on the signedness, and that will largely improve ergonomics as well. We don't think adding more overloads to texture functions is the right path.

**TL;DR**: We acknowledge a problem to be solved. We propose to change all of the signed arguments (that aren't supposed to be negative) into **unsigned** (with an exception of array indexing, which accepts both today, but we can talk about that). We don't think any overrides should be added for integer signs of texture functions.
